pinot-core/src/main/java/org/apache/pinot/core/data/table/SortedRecordTable.java

```diff
@@ -0,0 +1,340 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.table;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.core.operator.blocks.results.GroupByResultsBlock;
+import org.apache.pinot.core.query.aggregation.function.AggregationFunction;
+import org.apache.pinot.core.query.request.context.QueryContext;
+import org.apache.pinot.core.query.scheduler.resources.ResourceManager;
+import org.apache.pinot.core.util.QueryMultiThreadingUtils;
+import org.apache.pinot.core.util.trace.TraceCallable;
+
+
+/**
+ * Util class used for merging of sorted group-by aggregation
+ */
+public class SortedRecordTable extends BaseTable {
+  private final ExecutorService _executorService;
+  protected final int _resultSize;
+  protected final int _numKeyColumns;
+  protected final AggregationFunction[] _aggregationFunctions;
+
+  protected Record[] _topRecords;
+
+  private Record[] _records;
+  private int _numMergedBlocks;
+  private final int _desiredNumMergedBlocks;
+  private int _nextIdx;
+  private final Comparator<Record> _comparator;
+  protected final int _numThreadsExtractFinalResult;
+  protected final int _chunkSizeExtractFinalResult;
+
+  public SortedRecordTable(DataSchema dataSchema, QueryContext queryContext, int resultSize,
+      ExecutorService executorService,
+      int numMergedBlocks, int desiredNumMergedBlocks, Comparator<Record> comparator) {
+    super(dataSchema);
+    _numMergedBlocks = numMergedBlocks;
+    _desiredNumMergedBlocks = desiredNumMergedBlocks;
+    _numKeyColumns = queryContext.getGroupByExpressions().size();
+    _aggregationFunctions = queryContext.getAggregationFunctions();
+    _executorService = executorService;
+    _comparator = comparator;
+    _resultSize = resultSize;
+    _numThreadsExtractFinalResult = Math.min(queryContext.getNumThreadsExtractFinalResult(),
+        Math.max(1, ResourceManager.DEFAULT_QUERY_RUNNER_THREADS));
+    _chunkSizeExtractFinalResult = queryContext.getChunkSizeExtractFinalResult();
+    _records = new Record[_resultSize];
+    _nextIdx = 0;
+  }
+
+  ///  only used when creating SortedRecordTable from unique, sorted segment groupby results
```

Comment:
(minor, format) Keep one space after `///`, also we usually capitalize the javadoc. Same for other places